src/backend/gitalytics_api/common.py

```diff
@@ -95,3 +98,27 @@ def __init__(self, bearer: str):
     def auth_flow(self, request: httpx.Request) -> t.Generator[httpx.Request, httpx.Response, None]:
         request.headers["Authorization"] = self._auth_header
         yield request
+
+
+def add_error_logging(*, reraise_exception: bool):
+    def decorator(function):
+        if inspect.iscoroutine(function):
+            @functools.wraps(function)
+            async def wrapper(*args, **kwargs):
+                try:
+                    return await function(*args, **kwargs)
+                except Exception as exception:
+                    logging.exception(f"{function} raised an {exception.__class__.__name__}", exc_info=exception)
+                    if reraise_exception:
+                        raise
```

Comment:
```suggestion
                        raise exception
```
ist das nicht auch notwendig?

Reply:
nein, nicht notwendig, `raise` ohne eine neue bzw die alte exception gilt als "re-raise" und macht, dass die aktuelle exception durchgelassen (nicht gefangen) wird (deins könnte man aber theoretisch auch machen)